From foo@baz.org Wed Nov 16 13:31:19 2005
Date: Wed, 16 Nov 2005 13:41:28 -0800
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: USB: add dynamic id functionality to USB serial drivers

This lets the usb-serial drivers work properly with dynamic ids.

Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>


---
 drivers/usb/serial/pl2303.c     |    4 ++--
 drivers/usb/serial/usb-serial.c |    7 +++++++
 2 files changed, 9 insertions(+), 2 deletions(-)

--- gregkh-2.6.orig/drivers/usb/serial/pl2303.c
+++ gregkh-2.6/drivers/usb/serial/pl2303.c
@@ -58,8 +58,8 @@ static struct usb_device_id id_table [] 
 	{ USB_DEVICE(PL2303_VENDOR_ID, PL2303_PRODUCT_ID_RSAQ3) },
 	{ USB_DEVICE(PL2303_VENDOR_ID, PL2303_PRODUCT_ID_PHAROS) },
 	{ USB_DEVICE(IODATA_VENDOR_ID, IODATA_PRODUCT_ID) },
-	{ USB_DEVICE(ATEN_VENDOR_ID, ATEN_PRODUCT_ID) },
-	{ USB_DEVICE(ATEN_VENDOR_ID2, ATEN_PRODUCT_ID) },
+//	{ USB_DEVICE(ATEN_VENDOR_ID, ATEN_PRODUCT_ID) },
+//	{ USB_DEVICE(ATEN_VENDOR_ID2, ATEN_PRODUCT_ID) },
 	{ USB_DEVICE(ELCOM_VENDOR_ID, ELCOM_PRODUCT_ID) },
 	{ USB_DEVICE(ELCOM_VENDOR_ID, ELCOM_PRODUCT_ID_UCSGT) },
 	{ USB_DEVICE(ITEGNO_VENDOR_ID, ITEGNO_PRODUCT_ID) },
--- gregkh-2.6.orig/drivers/usb/serial/usb-serial.c
+++ gregkh-2.6/drivers/usb/serial/usb-serial.c
@@ -540,6 +540,7 @@ static struct usb_serial_driver *search_
 	struct list_head *p;
 	const struct usb_device_id *id;
 	struct usb_serial_driver *t;
+	struct usb_driver *usb_driver = to_usb_driver(iface->dev.driver);
 
 	/* List trough know devices and see if the usb id matches */
 	list_for_each(p, &usb_serial_driver_list) {
@@ -551,6 +552,12 @@ static struct usb_serial_driver *search_
 		}
 	}
 
+	id = usb_match_dynamic_id(iface, usb_driver);
+	if (id != NULL) {
+		dbg("dynamic descriptor matches");
+		return t;
+	}
+
 	return NULL;
 }
 
